File: library-core/src/main/java/com/mikepenz/fastadapter/utils/DefaultIdDistributorImpl.java
package com.mikepenz.fastadapter.utils;

import com.mikepenz.fastadapter.IIdentifyable;

import java.util.concurrent.atomic.AtomicLong;

/**
 * Created by mikepenz on 19.09.15.
 */
public class DefaultIdDistributorImpl<Identifiable extends IIdentifyable> extends DefaultIdDistributor<Identifiable> {

    private final AtomicLong idDistributor = new AtomicLong(-2L);

    @Override
    public long nextId(Identifiable identifiable) {
        return idDistributor.decrementAndGet();
    }
}

File: fastadapter/src/main/java/com/mikepenz/fastadapter/utils/DefaultIdDistributorImpl.kt
package com.mikepenz.fastadapter.utils

import com.mikepenz.fastadapter.IIdentifiable
import java.util.concurrent.atomic.AtomicLong

/**
 * Created by mikepenz on 19.09.15.
 */
class DefaultIdDistributorImpl<Identifiable : IIdentifiable> :
        DefaultIdDistributor<Identifiable>() {
    private val idDistributor = AtomicLong(-2L)

    override fun nextId(identifiable: Identifiable): Long {
        return idDistributor.decrementAndGet()
    }
}
